dagger-functions/src/main/java/io/odpf/dagger/functions/transformers/SQLTransformer.java

```diff
@@ -53,8 +52,7 @@ public StreamInfo transform(StreamInfo inputStreamInfo) {
             schema = schema.replace(ROWTIME, ROWTIME + ".rowtime");
             inputStream = assignTimeAttribute(inputStream);
         }
-        StreamExecutionEnvironment streamExecutionEnvironment = inputStream.getExecutionEnvironment();
-        StreamTableEnvironment streamTableEnvironment = getStreamTableEnvironment(streamExecutionEnvironment);
+        StreamTableEnvironment streamTableEnvironment = Transformer.daggerContextGenerator().getTableEnvironment();
```

Comment:
So, `Transformer.daggerContextGenerator()` returns  a `DaggerContext` object. This object contains `io.odpf.dagger.common.configuration.Configuration` as a field. 

In addition, we also pass the same object `io.odpf.dagger.common.configuration.Configuration` in the constructor of `SQLTransformer`. Hence, we will have two references to the same object. 
Additionally, it doesn't look nice to add this method to the Transformer class as mentioned [here](https://github.com/odpf/dagger/pull/202/files#r992256461).

Is it possible to pass the `DaggerContext` instance to the `SQLTransformer` constructor itself  like this?

```
  public SQLTransformer(Map<String, String> transformationArguments, String[] columnNames, DaggerContext daggerContext) {
        this.columnNames = columnNames;
        this.sqlQuery = transformationArguments.get("sqlQuery");
        this.tableName = transformationArguments.getOrDefault("tableName", "data_stream");
        this.allowedLatenessInMs = Long.parseLong(transformationArguments.getOrDefault("allowedLatenessInMs", "0"));
    }
```


If you check the  `TransformProcessor.getTransformMethod()`

```
  protected Transformer getTransformMethod(TransformConfig transformConfig, String className, String[] columnNames) throws ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        Class<?> transformerClass = Class.forName(className);
        Constructor transformerClassConstructor = transformerClass.getConstructor(Map.class, String[].class, Configuration.class);
        return (Transformer) transformerClassConstructor.newInstance(transformConfig.getTransformationArguments(), columnNames, configuration);
    }
```

This is how we initialize each transformer. 
If you can pass this same `DaggerContext` object that we initialized in `KafkaProtoSQLProcessor` down to this class like this :

KafkaProtoSQLProcessor -> StreamManager -> PostProcessorFactory -> ParentPostProcessor -> TransformProcessor

This way we can also eliminate passing the `configuration` as it is already present in DaggerContext

Reply:
DaggerContext object is made available to `SQLTransformer` class through KafkaProtoSQLProcessor -> StreamManager -> PostProcessorFactory -> ParentPostProcessor -> TransformProcessor.  Also refactored some of the code in `PreProcessorFactory` and `PostProcessorFactory`